__tests__/api.test.js

```diff
@@ -207,5 +206,57 @@ describe("POST /api/articles/:article_id/comments", ()=>{
             });
 })
 
+describe("PATCH /api/articles/:article_id", ()=>{
+    test(`PATCH:200 finds an article by it's id and updates it by increasing the current article's vote property by 1`, () => {
+        const newVote = { inc_votes : 1 };
+        return request(app)
+        .patch('/api/articles/1')
+        .send(newVote)
+        .expect(200)
+        .then((response) => {
+            const article = response.body;
+            expect(article.article_id).toBe(1)
+            expect(article.votes).toBe(101)
+            expect(article.title).toBe('Living in the shadow of a great man')
+        });
+    });
+    test(`PATCH:200 finds an article by it's id and updates it by would decreasing the current article's vote property by 100`, () => {
+        const newVote = { inc_votes : -100 };
+        return request(app)
+        .patch('/api/articles/3')
+        .send(newVote)
+        .expect(200)
+        .then((response) => {
+            const article = response.body;
+            expect(article.article_id).toBe(3)
+            expect(article.votes).toBe(-100)
+            expect(article.title).toBe('Eight pug gifs that remind me of mitch')
+        });
+    });
+    test('PATCH:404 sends an appropriate status and error message when given a valid but non-existent id', () => {
+        const newVote = { inc_votes : 50 };
+        return request(app)
+        .patch('/api/articles/999999')
+        .send(newVote)
+        .expect(404)
+        .then((response) => {
+            const article = response.body;
+            expect(article.msg).toBe('No user found for user_id: 999999');
+        });
+    });
+    test('PATCH:400 sends an appropriate status and error message for missing keys in post object', () => {
+        const newVote = {  };
+        return request(app)
+        .patch('/api/articles/idnumbertwo')
+        .send(newVote)
+        .expect(400)
+        .then((response) => {
+            const article = response.body;
+            expect(article.msg).toBe('Bad request');
+        });
+    });
```

Comment:
There's two things wrong with this request that would cause it to error out
- The ID is not a number (400)
- The inc_votes property is missing
When testing our error handling, we should be sure to only break one thing at a time! Otherwise we won't be sure what actually causes the error

Also, interestingly enough, if only the inc_votes is missing, we can consider this a 200 and just return the article without update the votes. This leaves room for us later on if we wanted to add anything to the patch request. You could count it as a 400 instead, but this makes the api be a bit harder to work on if you wanted any extra features